### DISCUSSION TITLE
Being able to view anchor IDs in Tdoa2 , which the tag uses to calculate its position.

### DISCUSSION THREAD

**Discussion by DavideLentini** [bitcraze/discussions#619] [General]
Hello everyone , I don't know if this is the right place to ask this question , I hope so ! 
My tag uses bitcraze firmware , I have a system with 8 UWB anchors running in Tdoa2 . My system is working and I also got some decent results . Now , since I would like to improve the system , I would need to understand , which (among the 8 anchors) the tag chooses to calculate its _X,Y,Z_ position . Once understood internally in the code , I should pass this data , but that is another matter . For example : _the tag calculated that its position is x=4,y=5,z=1 and used the information/packages from these anchors to calculate it : id=3,id=5,id=7_. This is my idea , I don't know if it is possible or where I have to act in the code . Thanks anyway for the help !

**Comment by krichardsson** [comment]
What you are looking for is probably the [`findSuitableAnchor()`](https://github.com/bitcraze/crazyflie-firmware/blob/master/src/utils/src/tdoa/tdoaEngine.c#L193-L213) function.
This function is called for an incoming packet to match it with the packets that have been received earlier, to do the TDoA computation. 

Currently there are two options: to match with youngest (most recent) packet or with a random packet. 
In theory it would be possible to optimize based on the geometry of the system, radio quality or other properties instead.

I have not seen any substantial differences in my tests, I have not put that much time into it on the other hand.
